src/Stryker.Core/Stryker.Core/Mutants/MutantOrchestrator.cs

```diff
@@ -62,12 +64,150 @@ public IEnumerable<Mutant> GetLatestMutantBatch()
+            foreach (var child in children)
+            {
+                var originalNode = childCopy.GetCurrentNode(child);
+                var mutatedNode = Mutate(child);
+                if (!mutatedNode.IsEquivalentTo(originalNode))
+                {
+                    childCopy = childCopy.ReplaceNode(originalNode, mutatedNode);
+                }
+            }
+            return childCopy;
+        }
+
+        private SyntaxNode MutateExpressionStatement(ExpressionStatementSyntax expressionStatement)
+        {
+            if (expressionStatementNeedingIf.Contains(expressionStatement.Expression.GetType()))
+            {
+                var expressionCopy = expressionStatement.TrackNodes(expressionStatement, expressionStatement.Expression);
+                SyntaxNode result = MutateSubExpressionWithIfStatements(expressionStatement, expressionCopy, expressionStatement.Expression);
+                if (!(expressionStatement.Expression is AssignmentExpressionSyntax))
+                {
+                    return result;
+                }
+
+                return result.ReplaceNode(result.GetCurrentNode(expressionStatement), MutateExpressions(expressionStatement));
+            }
+
+            return MutateExpressions(expressionStatement);
+        }
+
+
+        private SyntaxNode MutateIfStatement(IfStatementSyntax ifStatement)
+        {
+            var mutatedIf = ifStatement.Else != null
+                ? ifStatement.TrackNodes(ifStatement.Condition, ifStatement.Statement, ifStatement.Else)
+                : ifStatement.TrackNodes(ifStatement.Condition, ifStatement.Statement);
+
+            if (!ifStatement.Condition.DescendantNodes().Any(x =>
+                x.IsKind(SyntaxKind.DeclarationExpression) || x.IsKind(SyntaxKind.DeclarationPattern)))
+            {
+                mutatedIf = mutatedIf.ReplaceNode(mutatedIf.GetCurrentNode(ifStatement.Condition),
+                    MutateWithConditionalExpressions(ifStatement.Condition));
+            }
+
+            if (ifStatement.Else != null)
+            {
+                mutatedIf = mutatedIf.ReplaceNode(mutatedIf.GetCurrentNode(ifStatement.Else), Mutate(ifStatement.Else));
+            }
+
+            return mutatedIf.ReplaceNode(mutatedIf.GetCurrentNode(ifStatement.Statement), Mutate(ifStatement.Statement));
+        }
+
+        private SyntaxNode MutateForStatement(ForStatementSyntax forStatement)
+        {
+            // for needs special treatments for its incrementors
+            StatementSyntax trackedCopy = forStatement;
+
+            foreach (var incrementor in forStatement.Incrementors)
+            {
+                trackedCopy = MutateSubExpressionWithIfStatements(forStatement, trackedCopy, incrementor);
+            }
+
+            var originalFor = trackedCopy;
+            if (trackedCopy != forStatement)
+            {
+                var scan = trackedCopy;
+                while (scan is IfStatementSyntax mutantIf)
```

Comment:
What is going on here?

Reply:
I need to traverse through the ifs to locate the for that needs further mutation. I don't like it either, but I was not able to find any other way (including tracknodes)